File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/cadastro/usuario/UsuarioDTO.java
package br.com.grupopipa.gestaointegrada.cadastro.usuario;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.cadastro.perfil.PerfilDTO;
import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class UsuarioDTO implements DTO {

    private UUID id;
    private String nome;
    private String login;
    private String senha;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
    private List<PerfilDTO> perfis;

}

File: src/frontend/src/app/components/cadastro/usuario/model/usuario-dto.ts
import { PerfilDTO } from '../../perfil/model/perfil-dto';
import { UsuarioUnidadeNegocioDTO } from './usuario-unidade-negocio-dto';

export class UsuarioDTO {
  id: string;
  nome: string;
  login: string;
  senha: string;
  createdat?: Date;
  updatedAt?: Date;
  createdBy?: string;
  updatedBy?: string;
  perfis?: PerfilDTO[];
  unidadesNegocio?: UsuarioUnidadeNegocioDTO[];

  constructor(
    id: string,
    nome: string,
    login: string,
    senha: string,
    createdat?: Date,
    updatedAt?: Date,
    createdBy?: string,
    updatedBy?: string,
    perfis?: PerfilDTO[],
    unidadesNegocio?: UsuarioUnidadeNegocioDTO[]
  ) {
    this.id = id;
    this.nome = nome;
    this.login = login;
    this.senha = senha;
    this.createdat = createdat;
    this.updatedAt = updatedAt;
    this.createdBy = createdBy;
    this.updatedBy = updatedBy;
    this.perfis = perfis;
    this.unidadesNegocio = unidadesNegocio;
  }
}
